testing/python/approx.py

```diff
@@ -466,18 +476,48 @@ def test_foo():
     @pytest.mark.parametrize(
         "x",
         [
-            pytest.param(None),
-            pytest.param("string"),
-            pytest.param(["string"], id="nested-str"),
             pytest.param([[1]], id="nested-list"),
-            pytest.param({"key": "string"}, id="dict-with-string"),
             pytest.param({"key": {"key": 1}}, id="nested-dict"),
         ],
     )
     def test_expected_value_type_error(self, x):
         with pytest.raises(TypeError):
             approx(x)
 
+    @pytest.mark.parametrize(
+        "x",
+        [
+            pytest.param(None),
+            pytest.param("string"),
+            pytest.param(["string"], id="nested-str"),
+            pytest.param({"key": "string"}, id="dict-with-string"),
+        ],
+    )
+    def test_nonnumeric_okay_if_equal(self, x):
```

Comment:
We should include another test which also checks the message when equality fails:

```python
def test():
    x1 = {"foo": 1.0000005, "bar": None, "foobar":1}
    x2 = {"foo": 1.0000006, "bar": None}
    assert x1 == pytest.approx(x2)
```

This currently outputs:

```
_______________________________ test ________________________________

    def test():
        x1 = {"foo": 1.0000005, "bar": None, "foobar":1}
        x2 = {"foo": 1.0000006, "bar": None}
>       assert x1 == pytest.approx(x2)
E       assert {'bar': None, 'foo': 1.0000005, 'foobar': 1} == <[TypeError("bad operand type for abs(): 'NoneType'") raised in repr()] ApproxMapping object at 0x281cf9f3430>
E        +  where <[TypeError("bad operand type for abs(): 'NoneType'") raised in repr()] ApproxMapping object at 0x281cf9f3430> = <function approx at 0x00000281CECB3CA0>({'bar': None, 'foo': 1.0000006})
E        +    where <function approx at 0x00000281CECB3CA0> = pytest.approx
```

That message should be improved. 👍

Reply:
Right, I did not think to check the message. Should be fixed now.